octoprint_mrbeam/static/js/helpers/working_area_helper.js

```diff
@@ -1,4 +1,48 @@
 class WorkingAreaHelper {
+    static versionCompare(v1, v2, options) {
+        var lexicographical = options && options.lexicographical,
+            zeroExtend = options && options.zeroExtend,
+            v1parts = v1.split("."),
+            v2parts = v2.split(".");
+
+        function isValidPart(x) {
+            return (lexicographical ? /^\d+[A-Za-z]*$/ : /^\d+$/).test(x);
+        }
+
+        if (!v1parts.every(isValidPart) || !v2parts.every(isValidPart)) {
+            return NaN;
```

Comment:
I am also surprised this is not crashing the comparison... 🤔  NaN <= 0 returns false for some reason which escapes my understanding... Lovely JavaScript....

Reply:
also NaN >= 0 is false... In any case I think we shouldn't return 2 different types from one function... 😓

Reply:
Yea while Ahmed and I were going through this we found out that if the versions looks like: "1.1.2 (1:1.1+202202050950+0a00cf5339)" then this method will always return NaN which after applied to: NaN <= 0 it will always return false meaning all Inkscape designs passing through this has the new version scaling all the time.